src/main/java/com/openshift/internal/restclient/model/volume/VolumeSource.java

```diff
@@ -14,22 +14,22 @@
 
 import java.util.HashMap;
 
+import com.openshift.restclient.model.volume.VolumeType;
 import org.jboss.dmr.ModelNode;
 
 import com.openshift.internal.restclient.model.ModelNodeAdapter;
 import com.openshift.internal.restclient.model.properties.ResourcePropertyKeys;
-import com.openshift.restclient.model.volume.IVolume;
 import com.openshift.restclient.model.volume.IVolumeSource;
 
 /**
  * 
  * @author Jeff Cantrill
  *
  */
-public class VolumeSource 
+public abstract class VolumeSource
```

Comment:
The volume sources has no metadata, kind or anything like that like a resource has. Its not defined as a resource by the REST api, see https://docs.openshift.org/latest/rest_api/kubernetes_v1.html#v1-volume and https://docs.openshift.org/latest/rest_api/kubernetes_v1.html#v1-emptydirvolumesource for instance.

There is PersistentVolumeClaim resource, but that should be a different interface, and already exists in the form of IPersistentVolumeClaim. This is more like a reference to that resource.